### ISSUE TITLE
Plugins implementing `FunctionReturnTypeProviderInterface` are not triggered anymore

### ISSUE THREAD

**Issue by boesing** [vimeo/psalm#10275]
Starting with v5.13.0, plugins providing `FunctionReturnTypeProviderInterface` are not properly triggered anymore.

To reproduce this, create a plugin like this:

```php
final class ReturnTypeProvider implements FunctionReturnTypeProviderInterface
{
    private const FUNCTION_SPRINTF = 'sprintf';

    /**
     * @psalm-return non-empty-list<lowercase-string>
     */
    public static function getFunctionIds(): array
    {
        return [self::FUNCTION_SPRINTF];
    }

    public static function getFunctionReturnType(FunctionReturnTypeProviderEvent $event): ?Type\Union
    {
        die('here');
    }
}

final class Plugin implements PluginEntryPointInterface
{
    public function __invoke(RegistrationInterface $registration, ?SimpleXMLElement $config = null): void
    {
        $registration->registerHooksFromClass(ReturnTypeProvider::class);
    }
}
```

**Comment by boesing** [comment]
Oh, as I can see, they are actually not triggered in case some other return type provider was providing a return type.
It seems that with 5.13.0, an `sprintf` return type provider was added and therefore, the `FunctionReturnTypeProvider` is early returning...

Any ideas on how plugins can get a higher priority than those handlers registered in the `__construct`?
My plugin does actually provide a more precise return type for `sprintf`. 

Maybe `FunctionReturnTypeProvider::registerClosure` could use `array_unshift`?
Thoughts?

**Comment by boesing** [comment]
I prefer plugins as they are somewhat opionated and having this part of psalm itself would shift anything regarding issues to this repository which then also means maintenance. 

I can add some parts to psalm but overall I'd prefer plugins having more power over internal stuff is why plugins exist in the first place. 🤔

So I'd be fine with migrating some stuff to psalm while having my plugin for some more specific/opinionated/to be tested stuff on top.
